### PR TITLE
Fix #198 add pkexcept_to_file, pykern test empties work_dir

### LINKED ISSUE #198: case_dirs support deviance cases
@gurhar1133 I need this for something I'm working on so I'll prototype something and maybe have you take it over.

### PR DESCRIPTION
**Pull request by robnagler**
case_dirs no longer calls empty_work_dir so all the dirs are around at the end of the test.

@gurhar1133 note the use of pkexcept_to_file() in fmt_test.py and the way the regexp was implemented to avoid problems. It does mean that the test has to run from a "HOME" directory, but that's a likely case. We'll see if we get complaints.

### CODE REVIEW

**Review by gurhar1133 — COMMENTED** [review]
Check failed with:
```
File "/home/runner/work/pykern/pykern/pykern/pkcli/test.py", line 126, in _remove_work_dir
    w = _TEST_PY.sub(pkunit.WORK_DIR_SUFFIX, test_file)
NameError: name 'pkunit' is not defined
```

LMK when check passes and I'll take another look 👍

File: pykern/pkcli/test.py [hunk 1]
```diff
@@ -47,6 +52,7 @@ def default_command(*args):
     for t in paths:
         n += 1
         o = t.replace('.py', '.log')
+        pkio.unchecked_remove(_TEST_PY.sub(pkunit.WORK_DIR_SUFFIX, t))
```
[the remark this answers is not in the document]
Comment (robnagler) [reply pykern/pkcli/test.py]: Test passes.

> Just a question, but why dont we use string .replace() for simple patterns like this?

This is the regex: https://github.com/radiasoft/pykern/pull/199/files#diff-e7f5dd4d1c16b1812b07e3b26e36c0632186e89b3afcda650252ccc144327ba0R15

[the remark this answers is not in the document]
Comment (gurhar1133) [reply pykern/pkcli/test.py]:
```
File "/home/runner/work/pykern/pykern/pykern/pkcli/test.py", line 126, in _remove_work_dir
    w = _TEST_PY.sub(pkunit.WORK_DIR_SUFFIX, test_file)
NameError: name 'pkunit' is not defined
```
Checks not passing ^

[the remark this answers is not in the document]
Comment (robnagler) [reply pykern/pkcli/test.py]: Ugh, committed to wrong branch. Fixed now.

**Review by gurhar1133 — APPROVED** [review]
Looks good. `pkexcept_to_file` should be pretty useful. I have some thoughts about the .in/.out data_dir() structure that I would like to chat about some time, but that is a separate issue

**Pull request merged**